src/vendor/perfherder.js

```diff
@@ -260,11 +250,33 @@ return a list of {meta, data} objects, each representing
 a perfhereder signature that matches the given filter
  */
 const getData = async (condition) => {
-  const signatures = await getSignatures(condition);
+  const collated = extract(condition, ['push_timestamp', 'repo', 'framework']);
+
+  const results = await promiseAll(collated
+    .map(async ([pushDate, repo, framework, rest]) => {
+      if (repo === true || framework === true) {
+        Log.error('expecting expression to have both repo and framework');
+      }
+      await getFramework({ repo: repo.eq.repo, framework: framework.eq.framework });
+
+      Log.note('scan {{num}} signatures', { num: PERFHERDER.signatures.length });
+
+      const data = await getDataBySignature(PERFHERDER.signatures.filter(jx(rest)));
```

Comment:
@gmierz the second step is to filter the signatures by the residual expression: It is interesting to note that data filtering  is now faster because the residue expression is simpler, and executes faster over the 50K signatures.